### PR TITLE
Add a test that verifies we prefer a non lexical scope over a lexical scope when doing unqualified name lookup

### PR DESCRIPTION
**Pull request by bricknerb**
This adds missing test coverage.
See discussion in https://github.com/carbon-language/carbon-lang/pull/4574.

### CODE REVIEW

**Comment by geoffromer** [comment]
Since this came out of #4574, zygoloid might be a better reviewer.

File: toolchain/check/testdata/function/call/prefer_unqualified_lookup.carbon [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/function/call/prefer_unqualified_lookup.carbon
+
+// --- prefer_unqualified_lookup.carbon
+
+library "[[@TEST_NAME]]";
+
+// F in the lexical scope.
+class Class(F:! type) {
+  class Inner {
+    // F in a non-lexical scope.                      ^
```
Comment (zygoloid) [inline toolchain/check/testdata/function/call/prefer_unqualified_lookup.carbon]:
```suggestion
    // F in a non-lexical scope.
```
Did you mean to include this caret?

**Review by zygoloid — APPROVED** [review]
Looks good, thanks!

**Comment by zygoloid** [comment]
(It looks like you need to rerun `autoupdate_testdata` -- the test expectations don't quite match the output. Usually this happens because line numbers in the test changed since the previous autoupdate run; sometimes autoupdate takes two runs to stabilize for this reason.)

File: toolchain/check/testdata/function/call/prefer_unqualified_lookup.carbon (hunk 1 above)
Comment (bricknerb) [reply toolchain/check/testdata/function/call/prefer_unqualified_lookup.carbon]: Nope.
Done.

**Comment by bricknerb** [comment]
> (It looks like you need to rerun `autoupdate_testdata` -- the test expectations don't quite match the output. Usually this happens because line numbers in the test changed since the previous autoupdate run; sometimes autoupdate takes two runs to stabilize for this reason.)

I hope this passes now.
It definitely passes locally and it seems like I need some stamp here to make it run here.

**Pull request merged**